Frameworks/Foundation/NSOperation.mm

```diff
@@ -131,139 +128,109 @@ - (void)setThreadPriority:(double)priority {
  @Status Interoperable
 */
 - (void)setCompletionBlock:(void (^)(void))block {
-    id oldBlock = priv->completionBlock;
-    priv->completionBlock = [block copy];
-    [oldBlock release];
+    pthread_mutex_lock(&_completionBlockLock);
+    [_completionBlock autorelease];
+    _completionBlock = [block copy];
+    pthread_mutex_unlock(&_completionBlockLock);
 }
 
 /**
  @Status Interoperable
 */
 - (void (^)(void))completionBlock {
```

Comment:
needs
    [self willChangeValueForKey:@"completionBlock"];
    [self didChangeValueForKey:@"completionBlock"];


---
In reply to: [69357300](https://github.com/Microsoft/WinObjC/pull/602#discussion_r69357300) [](ancestors = 69357300)